azure-client-authentication/src/main/java/com/microsoft/azure/v2/credentials/MSICredentials.java

```diff
@@ -18,41 +16,145 @@
 import java.io.OutputStreamWriter;
 import java.net.HttpURLConnection;
 import java.net.URL;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 
 /**
  * Managed Service Identity token based credentials for use with a REST Service Client.
  */
 @Beta
-public class MSICredentials extends AzureTokenCredentials {
-    private final String resource;
-    private final int msiPort;
-    private final AzureJacksonAdapter adapter;
+public final class MSICredentials extends AzureTokenCredentials {
+    //
+    private final List<Integer> retrySlots = new ArrayList<>(Arrays.asList(new Integer[] {1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765}));
```

Comment:
Wanted to ask when you were OOF - why Fibonacci here?

Reply:
Last release we made a patch to allow setting max retry - but max retry can go out of bound in this array. An exponential retry can avoid that and also a default max retry other than 6700 seconds may be a little more user-friendly

Reply:
Agreed, let's switch to exponential and will take care of slot array OOB as well.